app/src/main/java/org/greenstand/android/TreeTracker/view/TextButton.kt

```diff
@@ -415,7 +470,7 @@ fun DepthButton(
                     if (!isEnabled) return@pointerInput
                     detectTapGestures(
                         onTap = {
-                            onClickState.value.invoke()
+                            onClick()
```

Comment:
reverse this also and leave the previous line. This change would cause the depthButton to always use default state and won't recreate it. For more info see #1000 and #984

Reply:
Reverted. Thanks!